rotkehlchen/tests/exchanges/test_woo.py

```diff
@@ -0,0 +1,137 @@
+            'side': 'SELL',
+            'fee': 0.2,
+            'fee_asset': 'USDT',
+            'executed_timestamp': '1634610000.0',
+        },
+    ]
+    with patch.object(Woo, 'query_online_trade_history', return_value=mock_trades):
+        result = mock_woo.query_online_trade_history(
+            start_ts=Timestamp(start_ts),
+            end_ts=Timestamp(end_ts),
+        )
+    assert result == mock_trades
+
+
+@pytest.mark.parametrize(('start_ts', 'end_ts'), [(0, 1), (1634600000, 1634620000)])
+def test_query_online_history_paginated(mock_woo, start_ts, end_ts):
+    with patch.object(Woo, '_api_query_paginated') as mock_api_query_paginated:
+        mock_woo.query_online_trade_history(
+            start_ts=Timestamp(start_ts),
+            end_ts=Timestamp(end_ts),
+        )
+        expected_call_page_one = call(
+            start_ts=Timestamp(start_ts),
+            end_ts=Timestamp(end_ts),
+            options={'limit': API_MAX_LIMIT},
+            case='trades',
+        )
+        assert mock_api_query_paginated.call_args == expected_call_page_one
+        mock_woo.query_online_trade_history(
+            start_ts=Timestamp(start_ts),
+            end_ts=Timestamp(end_ts),
+        )
+        expected_call_page_two = call(
+            start_ts=Timestamp(start_ts),
+            end_ts=Timestamp(end_ts),
+            options={'limit': API_MAX_LIMIT},
+            case='trades',
+        )
+        assert mock_api_query_paginated.call_args == expected_call_page_two
+
+
+def test_query_balances(mock_woo):
+    balances_response = {
+        'success': 'true',
+        'data': {
+            'holding': [
+                {
+                    'token': 'WOO',
+                    'holding': 1,
+                    'frozen': 0,
+                    'staked': 0,
+                    'unbonding': 0,
+                    'vault': 0,
+                    'interest': 0,
+                    'pendingShortQty': 0,
+                    'pendingLongQty': 0,
+                    'availableBalance': 0,
+                    'averageOpenPrice': 0.23432,
+                    'markPrice': 0.25177,
+                    'updatedTime': 312321.121,
+                },
+            ],
+        },
+    }
+
+    asset_balances = mock_woo._deserialize_accounts_balances(balances_response)
+    assert asset_balances == {
+        A_WOO: Balance(
+            amount=FVal('1'),
+            usd_value=FVal('1.5'),
+        ),
+    }
+
+
+def test_deserialize_trade_buy(mock_woo):
+    mock_trades = {
+        'id': 1,
+        'symbol': 'SPOT_BTC_ETH',
+        'order_id': 101,
+        'executed_price': 50000.0,
+        'executed_quantity': 1.0,
+        'side': 'BUY',
+        'fee': 0.1,
+        'fee_asset': 'ETH',
+        'executed_timestamp': '1634600000.0',
+    }
+    result = mock_woo._deserialize_trade(mock_trades)
+    assert result == Trade(
+        timestamp=1634600000,
+        location=Location.WOO,
+        base_asset=A_BTC,
```

Comment:
Can you also add a sell? If you read the docstring for the Trade class there is a game between quote/base and buy/sell